Rules/AvoidUnusableParameter.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+#if !CORECLR
+using System.ComponentModel.Composition;
+#endif
+using System.Globalization;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+    /// <summary>
+    /// AvoidUnusableParameter: Check if a parameter has a unusable name.
+    /// </summary>
+#if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+    public class AvoidUnusableParameter : IScriptRule
+    {
+        /// <summary>
+        /// AnalyzeScript: Check if a parameter has a unusable name.
+        /// </summary>
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+
+            // Finds all functionAst
+            IEnumerable<Ast> functionAsts = ast.FindAll(testAst => testAst is FunctionDefinitionAst, true);            
+            var regex = new Regex(@"^\d+$");

```

Comment:
please give this variable a better name what it does. e.g. `containsOnlyDigitsRegex`. Then you don't need the comment on line 46. Personally I am not a fan or regex. What about using `char.IsDigit`?

Reply:
Initial assumption (from the ticket) about digits only was wrong, i.e. `[switch]$1a` will be also unusable. I updated test to check if name starts with digit.